internal/auth0/connection/resource_scim.go

```diff
@@ -0,0 +1,88 @@
+package connection
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/auth0/terraform-provider-auth0/internal/config"
+	internalError "github.com/auth0/terraform-provider-auth0/internal/error"
+	internalSchema "github.com/auth0/terraform-provider-auth0/internal/schema"
+)
+
+// NewSCIMConfigurationResource will return a new auth0_scim_configuration (1:1) resource.
+func NewSCIMConfigurationResource() *schema.Resource {
+	return &schema.Resource{
+		Schema:        scimResourceSchema,
+		CreateContext: createSCIMConfiguration,
+		UpdateContext: updateSCIMConfiguration,
+		ReadContext:   readSCIMConfiguration,
+		DeleteContext: deleteSCIMConfiguration,
+		Importer: &schema.ResourceImporter{
+			StateContext: internalSchema.ImportResourceGroupID("client_id"),
+		},
+		Description: "With this resource, you can use `SCIM` to manage a connection.",
+	}
+}
+
+func createSCIMConfiguration(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*config.Config).GetAPI()
+	connectionID := data.Get("connection_id").(string)
+	scimConfiguration := expandSCIMConfiguration(data)
+
+	if err := api.Connection.CreateSCIMConfiguration(ctx, connectionID, scimConfiguration); err != nil {
+		return diag.FromErr(err)
+	}
+	internalSchema.SetResourceGroupID(data, connectionID)
+
+	return readSCIMConfiguration(ctx, data, meta)
+}
+
+func updateSCIMConfiguration(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*config.Config).GetAPI()
+	connectionID := data.Get("connection_id").(string)
+	scimConfiguration := expandSCIMConfiguration(data)
+
+	if err := api.Connection.UpdateSCIMConfiguration(ctx, connectionID, scimConfiguration); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return readSCIMConfiguration(ctx, data, meta)
+}
+
+func readSCIMConfiguration(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*config.Config).GetAPI()
+	connectionID := data.Get("connection_id").(string)
+
+	scimConfiguration, err := api.Connection.ReadSCIMConfiguration(ctx, connectionID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	defaultSCIMConfiguration, err := api.Connection.ReadSCIMDefaultConfiguration(ctx, connectionID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	data.SetId(connectionID)
+
+	result := multierror.Append(
+		flattenSCIMConfiguration(data, scimConfiguration),
+		data.Set("default_mapping", flattenSCIMMappings(defaultSCIMConfiguration.GetMapping())),
+	)
+
+	return diag.FromErr(result.ErrorOrNil())
+}
+
+func deleteSCIMConfiguration(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*config.Config).GetAPI()
+	connectionID := data.Get("connection_id").(string)
```

Comment:
```suggestion
	connectionID := data.Id()
```

Reply:
there were actually a few places in this change I could use that :-)